frontend/src/pages/contains-tracker-warning.page.tsx

```diff
@@ -49,6 +52,27 @@
       },
     });
 
+  const TrackerLinkWithMetrics = trackerData && (
+    <OutboundLink
+      to={trackerData.original_link}
+      eventLabel={l10n.getString("contains-tracker-warning-view-link-cta")}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      <span>{l10n.getString("contains-tracker-warning-view-link-cta")}</span>
+    </OutboundLink>
+  );
+
+  const TrackerLinkWithoutMetrics = trackerData && (
+    <Link
+      href={trackerData.original_link}
```

Comment:
CodeQL didn't realize that `react-ga`'s `OutboundLink` was a link, so we didn't get this warning on PR 3294. The code path is:

* The backend removes a tracker link from an email and replaces it with a link to this page. The original tracker link is embedded in the URL hash
* The Relay user clicks through to this page, which tells about the tracker, warning the user, and a button "Yes, view link"
* The Relay user clicks "Yes, view link" to view it.

The security issue is that someone could create a link to this page with their own URL, [like this](https://stage.fxprivaterelay.nonprod.cloudops.mozgcp.net/contains-tracker-warning/#{"sender":"sender@example.com","received_at":1707495386000,"original_link":"https://rroll.to/1x3YqZ"}).

My preference is to mark "Won't Fix" for now. There is a warning on the page to not click the link. If we do fix it, then it would be something like:

* Create a cryptographic signature of the original tracker URL on the server. We might be able to use something like JWT to help.
* Publish the public key so the JS can get it
* Verify the signature on the client side, do not create the link if signature validation fails.

This feels like a different effort than MPP-3644, and addresses an issue that has been around since May 2023.

Reply:
question (blocking): Since there's not a "Won't fix for now" button, can we think about how to address this later? Even if we simply file a ticket for a future sprint?

Reply:
I've filed https://mozilla-hub.atlassian.net/browse/MPP-3750 for this effort. Marking this as "Won't Fix".